public/app/features/dashboard/api/dashboard_api.ts

```diff
@@ -34,8 +35,8 @@ class LegacyDashboardAPI implements DashboardAPI {
     return getBackendSrv().delete<DeleteDashboardResponse>(`/api/dashboards/uid/${uid}`, { showSuccessAlert });
   }
 
-  getDashboardDTO(uid: string): Promise<DashboardDTO> {
-    return getBackendSrv().get<DashboardDTO>(`/api/dashboards/uid/${uid}`);
+  getDashboardDTO(uid: string, queryParams?: UrlQueryMap | undefined): Promise<DashboardDTO> {
+    return getBackendSrv().get<DashboardDTO>(`/api/dashboards/uid/${uid}`, queryParams);
```

Comment:
@bfmatei  - i think this PR has grown unnecessarily big given that you are changing the API for dashboard client. I think what would be a more reasonable change than passing `queryParams` all over the place, would be to implement the logic directly in this method that would have a minimal impact on the rest of the codebase:

```ts
 getDashboardDTO(uid: string): Promise<DashboardDTO> {
    if(passScopeToDashboardApi && hasQueryParammsInlocation) {
        return getBackendSrv().get<DashboardDTO>(`/api/dashboards/uid/${uid}`, scopeQueryParams);
    }
    
    return getBackendSrv().get<DashboardDTO>(`/api/dashboards/uid/${uid}`);
}
```

I'm afraid this change has a ripple effect that we may want to limit.

I'm also not sure how those query params would work in the k8s API that @ryantxu is working on (see K8sDashboardAPI), I think the k8s client `get` method accepts only uid ATM, so we need to figure out path for the new API as well.